eprei/ft_transcendence
# E play

Now the user can create a room or seek to join an existing one.

When the user creates a room, its status changes to 'playing'.
This status will need to be changed once the game ends or if the user gets tired of waiting for a second user to join and exits the waiting window.
I still have to make the waiting window as long as it will be displayed while the second user is not logged in.

When a user launches the 'matching system' its status temporarily changes to playing. If no room to join is found, the user's status is changed to 'online'.

Both the 'create room' button and the 'launch matching system' button check that the user's status is not 'playing'. If this is the case, they will throw a message warning the user that they are already playing in another tab or window.

To test this you can:

- launch matching system (should not work as there is no room created)

- 'create room' (this will only create a room in the backend but nothing will happen in the frontend)

- launch the matching system one more time (in this case nothing happens in the frontend but the user is assigned to the room he has previously created)

- if you try to create a room or launch the matching system again, there will be an error because the user is 'playing'


To check the status of the rooms at any time you can use Swagger.

You can also use:

```bash
curl \
  -X 'GET' \
  -H 'accept: application/json' \
  'http://localhost:8080/api/room'
```

Or use your favorite http request client and do a GET http://localhost:8080/api/room
